jwst/cube_build/cube_build_io_util.py

```diff
@@ -69,46 +69,117 @@ def read_cubepars(self, instrument_info):
-                # find the table entries for this combination
+            # find the table entries for this combination
             for tabdata in ptab.ifucubepars_table:
                 table_channel = tabdata['channel']
                 table_band = tabdata['band']
-                table_plt_scale = tabdata['PLTSCALE']
-                table_wresol = tabdata['WRESOL']
-                table_sroi = tabdata['SROI']
-                table_wroi = tabdata['WROI']
+                table_spaxelsize = tabdata['SPAXELSIZE']
+                table_spectralstep = tabdata['SPECTRALSTEP']
+                table_wavemin = tabdata['WAVEMIN']
+                table_wavemax = tabdata['WAVEMAX']
                 #match on this_channel and this_sub
-                if(compare_channel == table_channel and this_sub == table_band):
-                    instrument_info.SetSpatialScale(table_plt_scale,this_channel,this_sub)
-                    instrument_info.SetWaveRes(table_wresol,this_channel,this_sub)
-                    instrument_info.SetWaveROI(table_wroi,this_channel,this_sub)
-                    instrument_info.SetSpatialROI(table_sroi,this_channel,this_sub)
+                if(this_channel == table_channel and this_sub == table_band):
+                    instrument_info.SetSpatialSize(table_spaxelsize,this_channel,this_sub)
+                    instrument_info.SetSpectralStep(table_spectralstep,this_channel,this_sub)
+                    instrument_info.SetWaveMin(table_wavemin,this_channel,this_sub)
+                    instrument_info.SetWaveMax(table_wavemax,this_channel,this_sub)
 
+            for tabdata in ptab.ifucubepars_msn_table:
+                table_channel = tabdata['channel']
+                table_band = tabdata['band']
+                table_sroi = tabdata['ROISPATIAL']
+                table_wroi = tabdata['ROISPECTRAL']
+                table_power = tabdata['POWER']
+                table_softrad = tabdata['SOFTRAD']
+                #match on this_channel and this_sub
+                if(this_channel == table_channel and this_sub == table_band):
+                    instrument_info.SetSpatialROI(table_sroi,this_channel,this_sub)
+                    instrument_info.SetWaveROI(table_wroi,this_channel,this_sub)
+                    instrument_info.SetMSMPower(table_power,this_channel,this_sub)
+                    instrument_info.SetSoftRad(table_softrad,this_channel,this_sub)
+        for tabdata in ptab.ifucubepars_multichannel_wavetable:
+            table_wave = tabdata['WAVELENGTH']
+            table_sroi = tabdata['ROISPATIAL']
+            table_wroi = tabdata['ROISPECTRAL']
+            table_power = tabdata['POWER']
+            table_softrad = tabdata['SOFTRAD']
+            instrument_info.SetMultiChannelTable(table_wave,table_sroi,
+                                              table_wroi,table_power,
+                                              table_softrad)
+
+        print('Done reading cubepar reference file')
```

Comment:
Shouldn't use `print` statements. Always use some flavor of `log.xxxx` (info, debug, warning, ...)